Using submodule

hadisfr opened the pull request:
implement https://github.com/kevinxucs/Sublime-Gitignore/issues/5

**Please merge only one of this and https://github.com/kevinxucs/Sublime-Gitignore/pull/7.**

> **Note**: Some of old boilerplates are missing from [github/gitignore](https://github.com/github/gitignore). We may need to make a pull request to add them, before merge this, or include them separately in this project.

Noctis- committed:
Update comment

xeoneux committed:
Update VisualStudio.gitignore

shiftkey committed:
Merge pull request #2519 from xeoneux/patch-1

Update VisualStudio.gitignore

shiftkey committed:
Merge pull request #2517 from Noctis-/patch-1

Update comment

halbich committed:
Include Plugins folder

shiftkey committed:
Merge pull request #2524 from halbich/unreal-plugins-directories

UnrealEngine - Include Plugins folder

shiftkey committed:
Merge pull request #2516 from ButchDean/master

Ignore Visual Studio 2017 C++ app generated files.

shiftkey committed:
Merge pull request #2505 from bndw/go-remove-glide

Remove .glide/ ignore

shiftkey committed:
Merge pull request #2513 from lushc/feature/ignore-vendors

Revert "Remove Composer's file and directory."

Commit:
Update VisualStudio.gitignore (#2510)

* Update VisualStudio.gitignore

cmower committed:
TeX.gitignore modification for Emacs users (#2491)

* without . the auto directory is still considered for commits

* small addition that ensures all Emacs-Lisp files (.el files) are ignored

augustoproiete committed:
Add sccprj/ folder to global MATLAB.gitignore (#2493)

Commit:
added logs and devel directories to .gitignore (#2497)

thiminhnhut committed:
Add *.sta (standalone packages) for TeX (#2484)

matzar committed:
Update Unity.gitignore (#2502)

Prevents directories: `Library`
`Temp`
`Obj`
`Build`
`Builds`
`Assets/AssetStoreTools`
from being included in the remote if the project sits in the `Assets` folder.

jrreed committed:
Ignore Unity generated '*.pdb.meta' files for '*.pdb' files (#2486)

RyanGlScott committed:
Add cabal.project.local~ to Haskell.gitignore (#2479)

t-my committed:
Add ignore for Azure Stream Analytics local run output folder (#2498)

matzar committed:
Fixing .gitignore (#2531)

* Update Unity.gitignore

Prevents directories: `Library`
`Temp`
`Obj`
`Build`
`Builds`
`Assets/AssetStoreTools`
from being included in the remote if the project sits in the `Assets` folder.

* removed wildcards from Unity's .gitignore

tasken committed:
Update Jekyll.gitignore (#2540)

Add .jekyll-cache/ as per envygeeks/jekyll-assets

enricosada committed:
ignore MSBuild Binary and Structured Log (#2549)

ref http://msbuildlog.com/

richarth committed:
Ignoring the models builder out of date flag (#2543)

The out of date models flag is automatically generated by the system, we don't want this in Git

Commit:
Ignore node_modules (#2538)

Rails 5 uses npm to handle javascript dependancies, node_modules should not be checked in to git.

sllroberts committed:
Update Matlab.gitignore (#2533)

Added extensions for prepackaged app and toolbox file, pregenerated helpsearch folders.

Commit:
Add "cmake-build-release/" to Jetbrains.gitignore (#2552)

mostafasoufi committed:
Added Must Use Plugins to gotignore (#2553)

https://codex.wordpress.org/Must_Use_Plugins

timonsku committed:
Added a few common CAM file extensions (#2555)

wolf99 committed:
Add *.snk strong name key files (#2483)

* Add *.snk strong name key files

Strong name key files shouldn't be included in a repository AFAIK. they are intended to sign build output to verify that it comes from the correct publisher. having an *.snk in a repository would allow anyone to produce build as if they were the original publisher.

I guess some OSS projects might like to have *.snk files in their repos but that would be an exception to the rule.

* Make use of *.snk optional

Add note explaining use.

* Reduce explanation

Just use a link instead

rafiw committed:
Update Eclipse.gitignore (#2556)

* Update Eclipse.gitignore

* Update Eclipse.gitignore

TennyZhuang committed:
Ignore parcel-bundler cache

mxschmitt committed:
Go: Add '*.exe~' to .gitignore

When you build a Go executable, which does already exists and is in use on Windows (e.g. running) then the output binary name will be extended by a '~' prefix.
https://github.com/golang/go/blob/558eeb2d850d064b0b02b65a7bf3af6c108c248d/src/cmd/go/internal/work/exec.go#L1165-L1170

shiftkey committed:
Merge pull request #2561 from maxibanki/patch-1

Go: Add '*.exe~' to .gitignore

rgbkrk committed:
Update Node.gitignore

shiftkey committed:
Merge pull request #2562 from rgbkrk/patch-1

Update Node.gitignore

justinhartman committed:
New template file for Backup file formats

mbukejlovic committed:
Fixed the unignore of folder 'packages'

Considering the VisualStudio gitignore file contains a line "**/[Pp]ackages/*", these 2 lines from Umbraco gitignore never get applied properly to the desired files, even though Umbraco lines were added after the VS lines. I guess the VS's line is more specific, so it gets a priority of some sort.

shiftkey committed:
Merge pull request #2572 from mbukejlovic/patch-1

Fixed the unignore of folder 'packages'

jerzywie committed:
Add .nvuser NVidia Nsight configuration file

cyliang committed:
Remove two ignores for Python/Django

Both are neither python-specific nor django-specific:
- .static_storage
- .media

neetrath committed:
Update Android.gitignore

Add fastlane in Android gitignore

shiftkey committed:
Merge pull request #2578 from neetrath/patch-2

Update Android.gitignore

meneguzzi committed:
Update TeX.gitignore

Added a temporary file used in MacTeX

shiftkey committed:
Merge pull request #2579 from meneguzzi/patch-1

Update TeX.gitignore

raterus committed:
Add rptproj.rptuser

The VS 2017 version of the Business Intelligence project contains a user file (*.rptproj.rptuser) that should be ignored as well

felixdivo committed:
Added ignore for files created by `catkin_make_isolated`

Simply execute `catkin_make_isolated` and `build` and `devel` will have different names, but should be ignored as well.

Commit:
Added "ServiceFabricBackup/"

The latest update of Visual Studio 17 (15.5.5) introduces a new backup folder called "ServiceFabricBackup" before upgrading service fabric applications.

chalin committed:
Add `.dart_tool/`

Followup to https://github.com/dart-lang/site-www/issues/444#issuecomment-346560788

shiftkey committed:
Merge pull request #2593 from chalin/patch-1

Add `.dart_tool/`

shiftkey committed:
Merge pull request #2589 from komaflash/master

Added "ServiceFabricBackup/"

shiftkey committed:
Merge pull request #2587 from felixdivo/patch-1

Added ignore for files created by `catkin_make_isolated`

raimon49 committed:
Add ignore '.pytest_cache' directory in Unit test section

shiftkey committed:
Merge pull request #2594 from raimon49/add-ignore-cache-dir-in-unit-test-section

Add ignore '.pytest_cache' directory in Unit test section

Commit:
Fixes  WebStorm capitalization in comment

On line 1, WebStorm is capitalized the same way as the other IDE names.

shiftkey committed:
Merge pull request #2596 from nfcopier/patch-1

Fixes  WebStorm capitalization in comment

DKroot committed:
.idea/**/datasources.xml is now safe to store

Since IDEA 14 (2014), IDEA and other contemporaneous JebBrains IDEs don't store sensitive info in `datasources.xml` and it makes sense to version control to share project data sources among the team. `dataSources.local.xml` is user-specific: contain usernames and could be sensitive from a security perspective, albeit passwords could be externalized to e.g. Mac Keychain.

References: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839, https://youtrack.jetbrains.com/issue/IDEA-127105

shiftkey committed:
Merge pull request #2597 from DKroot/patch-1

.idea/**/datasources.xml is now safe to store

davidebeatrici committed:
Qt: Correct typo "QtCtreator" to "QtCreator"

dgabbe committed:
Exclude secret deployment key in rsconnect/

Exclude secret deployment key in rsconnect/, see https://shiny.rstudio.com/articles/shinyapps.html for details.

shiftkey committed:
Merge pull request #2603 from dgabbe/dgabbe-r-rsconnect

Exclude secret deployment key in rsconnect/

davidebeatrici committed:
Qt: Remove extra empty lines

shiftkey committed:
Merge pull request #2601 from davidebeatrici/qt-typo-fix

Qt: typo fix and extra empty lines removal

CKGrafico committed:
Remove typings ignore

Nobody is using v1 of typings anymore on new projects and 'typings' is the recommended name for the folders of your custom types.
On the other hand the official Visual Studio ignore is not ignoring this folder.

shiftkey committed:
Merge pull request #2608 from CKGrafico/patch-1

Remove typings ignore

josharian committed:
Joomla: don't ignore .gitignore

Root .gitignore files should generally be checked in.

GusGA committed:
Add {prod, dev, whatever_enviroment}.secret.exs

jtogrul committed:
Ignored default sqlite database of Django

fahall committed:
Create Godot.gitignore

A simple .gitignore for the Godot game engine.

BlueNalgene committed:
Added htlatex for TeX.gitignore

kevmoo committed:
Fix URL in comments

Points to valid URI

shiftkey committed:
Merge pull request #2617 from kevmoo/patch-1

Dart.gitignore: Fix URL in comments

shiftkey committed:
Merge pull request #2616 from BlueNalgene/LaTeX_htlatex

Added htlatex for TeX.gitignore

fahall committed:
remove system specific ignores.

TennyZhuang committed:
Merge branch 'master' into patch-2

patrickelectric committed:
KiCad: Add formar documentation link

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
KiCad: Add backup, cache and rascue formats

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

shiftkey committed:
Merge pull request #2624 from patrickelectric/kicad

Kicad: Add more formats and documentation link

shiftkey committed:
Merge pull request #2613 from fahall/master

Create Godot.gitignore

shiftkey committed:
Merge pull request #2611 from GusGA/master

Add {prod, dev, whatever_enviroment}.secret.exs in Elixir/Phoenix

shiftkey committed:
Merge pull request #2580 from raterus/patch-1

Add rptproj.rptuser

shiftkey committed:
Merge pull request #2576 from jerzywie/master

Add .nvuser NVidia Nsight configuration file

shiftkey committed:
Merge pull request #2577 from cyliang/master

Remove unreasonable ignores for Python/Django

shiftkey committed:
Merge pull request #2612 from jtogrul/master

Ignored default sqlite database of Django

shiftkey committed:
Merge pull request #2609 from josharian/master

Joomla: don't ignore .gitignore

schnoddelbotz committed:
Update documentation link cargo.toml vs cargo.lock

shiftkey committed:
Merge pull request #2625 from schnoddelbotz/update-rust-doc-link

Update documentation link cargo.toml vs cargo.lock

mariuz committed:
remove VS year from comments

shiftkey committed:
Merge pull request #2627 from mariuz/patch-1

remove VS year from comments

tirkarthi committed:
Update .gitignore with .cpcache

The current clojure deps tool creates .cpcache directory in the project directory while using .deps.edn

shiftkey committed:
Merge pull request #2629 from tirkarthi/patch-1

Update .gitignore with .cpcache

areyer committed:
allow .ssh directory

gotonode committed:
Update Node.gitignore

shiftkey committed:
Merge pull request #2639 from gotonode/patch-1

Update Node.gitignore

akelsch committed:
Remove colons at the end of comments

akelsch committed:
Use one hashtag instead of two

akelsch committed:
Remove unnecessary comment

shiftkey committed:
Merge pull request #2640 from akelsch/patch-1

Improve JetBrains.gitignore styling

linlux committed:
added: *.iobj, *.ipdb

"These files are produced when Incremental Link-Time Code Generation (LTCG) is enabled."
https://stackoverflow.com/questions/31554559/possible-to-stop-generating-ipdb-iobj-files-by-visual-studio-2015

chimame committed:
Add exclude credential master key of Rails 5.2

Neargye committed:
Add the idea caches to gitignore

shiftkey committed:
Merge pull request #2643 from chimame/add_exclude_credential_master_key_of_rails

Add exclude credential master key of Rails 5.2

ikemo3 committed:
JetBrains: add .idea/httpRequests

This directory used by HTTP Client in IntelliJ IDEA (and other product).
https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html#viewingResponse

shiftkey committed:
Merge pull request #2646 from ikemo3/jetbrains-http

JetBrains: add .idea/httpRequests

esaalii committed:
Update Android.gitignore

Ignore .idea/assetWizardSettings.xml for Android Studio 3.1

shiftkey committed:
Merge pull request #2644 from Terik23/ignore_idea_caches

Add the idea caches to gitignore

Commit:
Added default annotation processing directory to Eclipse.gitignore

eelregit committed:
Add latexrun default directory for auxiliary files

jkodroff committed:
Remove tfvars files from Terraform.gitignore

`tfvars` files should not be gitignore-d as they are used to parameterize a Terraform root for multiple environments, e.g.:

```bash
$ tree terraform/roots/vpc
terraform/roots/vpc
├── env
│   ├── production.tfvars
│   ├── qa.tfvars
│   └── staging.tfvars
└── main.tf
```

There may be a use case where a particular `tfvars` file should be ignored, but there's no naming convention I'm aware of to easily designate such a file.

twhitbeck committed:
Add shelf to JetBrains.gitignore

k4rtik committed:
Fix the anchor to ignore section

shiftkey committed:
Merge pull request #2653 from k4rtik/correct-link

Fix the anchor to ignore section

rivajunior committed:
Add a new .msix extension

Microsoft was announced a new extension of installer for Windows. I just added that one to be ignored as default, like others installers files.
References:
https://blogs.windows.com/buildingapps/2018/03/07/three-things-need-know-windows-developer-day/
https://github.com/Microsoft/msix-packaging

gzm55 committed:
maven-wrapper.jar can be removed from the source

see: https://github.com/takari/maven-wrapper#usage-without-binary-jar

shiftkey committed:
Merge pull request #2656 from rivajunior/patch-1

Add a new .msix extension

shiftkey committed:
Merge pull request #2642 from linlux/master

added: *.iobj, *.ipdb

matthewrdev committed:
Added MFractor's working folder to VisualStudio.gitignore

MFractor is a Xamarin/Visual Studio Mac productivity tool used by 1000's of Xamarin developers.

The `.mfractor/` folder should not be included in source control (but often is).

Commit:
Added *.nlo and *.nls extensions to TeX.gitignore

shiftkey committed:
Merge pull request #2659 from matthewrdev/patch-1

Added MFractor's working folder to VisualStudio.gitignore

shiftkey committed:
Merge pull request #2648 from leesah/master

Update Android.gitignore

shiftkey committed:
Merge pull request #2660 from jnslmk/master

Added *.nlo and *.nls extensions to TeX.gitignore

oscarandreu committed:
Added NiFi .nar files

shiftkey committed:
Merge pull request #2661 from oscarandreu/patch-1

Added Apache NiFi .nar files

shiftkey committed:
Merge pull request #2652 from twhitbeck/jetbrains-add-shelf

Add shelf to JetBrains.gitignore

juliancadi committed:
Specify screenshot files only, not whole folder

The whole screenshots folder shouldn't be ignore since there could be frameit configuration in it (Framefile.json, fonts, background.jpg, *.strings, etc...). It'd be better to just ignore the generated PNG screenshots.

koppor committed:
Ignore generated *.xmpi files

Source: https://texdoc.net/pkg/xmpincl -> page 3:

> basically create a new file .xmpi

shiftkey committed:
Merge pull request #2663 from koppor/patch-1

[TeX] Ignore generated *.xmpi files

juliancadi committed:
Double asterisk in path

shiftkey committed:
Merge pull request #2662 from juliancadi/master

[Fastlane] Specify screenshot files only, rather than whole folder (in case the user has frameit config)

redheness committed:
Visual studio cache issue

/.vs/ do not catch the visual studio cache folder but .vs/ do

shiftkey committed:
Merge pull request #2664 from kilian442/patch-1

Visual studio cache issue

juliancadi committed:
[Fastlane] Specify screenshot files only, rather than whole folder (in case the user has frameit config)

The whole screenshots folder shouldn't be ignore since there could be frameit configuration in it (Framefile.json, fonts, background.jpg, *.strings, etc...). It'd be better to just ignore the generated PNG screenshot files.

shiftkey committed:
Merge pull request #2666 from juliancadi/master

[Fastlane] Specify screenshot files only, rather than whole folder (i…

owenblacker committed:
Added *.opendb

thomaslima committed:
adding exception for *.spl from elsarticle.cls

shiftkey committed:
Merge pull request #2681 from owenblacker/patch-1

Added *.opendb

shiftkey committed:
Merge pull request #2683 from thomaslima/latex-elsarticle

adding exception for *.spl from elsarticle.cls

14roiron committed:
Exclude coder temporary folders

igorepst committed:
Add persistent undo to Vim.gitignore

shiftkey committed:
Merge pull request #2687 from igorepst/master

Add persistent undo to Vim.gitignore

nschonni committed:
Ignore Reporting Project Upgrade backup files

The conversion process from the old custom rptproj to the MSBuild based projects creates a .bak backup file

shiftkey committed:
Merge pull request #2688 from nschonni/patch-1

Ignore Reporting Project Upgrade backup files

jefrydco committed:
Add exception  for nuxt.js output

shastriUF committed:
Ignore the .cache folder for LabVIEW projects

Ignore the .cache folder for LabVIEW projects. This is a local folder with metadata used by LabVIEW NXG.

KenjiOhtsuka committed:
Update JetBrains.gitignore

ignore dbnavigator.xml

because it contains sensitive data, such as host, port, db, user name, and so on.

shiftkey committed:
Merge pull request #2696 from KenjiOhtsuka/feature/ignore_dbnavigator_config

Update JetBrains.gitignore

jkodroff committed:
Update Terraform.gitignore

shiftkey committed:
Merge pull request #2650 from jkodroff/patch-1

Remove tfvars files from Terraform.gitignore

shiftkey committed:
Merge pull request #2698 from eelregit/latexrun

Add latexrun default directory for auxiliary files

giggio committed:
Do not ignore launchsettings.json for VS

Ignoring `launchSettings.json` does not make much sense. Now .NET CLI even considers this file when running with `dotnet run`, as you can read [here](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-run?tabs=netcore2x).

This settings will be useful if shared among project members, so it should be commited to the repo.

Also, on the default `.gitignore` file generated by Visual Studio it is not ignored, so this causes confusion, as depending on how `.gitignore` was created it could be commited in or not.

shiftkey committed:
Merge pull request #2705 from giggio/patch-1

Do not ignore launchsettings.json for VS

darrylabbate committed:
Add all sourcemap file formats

segayuu committed:
Add Vuepress build defalult output Directory

Commit:
Ignore Terraform crash log files

shiftkey committed:
Merge pull request #2713 from segayuu/patch-1

Add Vuepress build defalult output Directory

shiftkey committed:
Merge pull request #2711 from rootbeersoup/master

Add all Sass sourcemap file formats

shiftkey committed:
Merge pull request #2714 from mastertinner/patch-1

Ignore Terraform crash log files

bcko committed:
Update Dart.gitignore

updated according to https://www.dartlang.org/guides/libraries/private-files

tejashah88 committed:
Add .serverless to gitignore

shiftkey committed:
Merge pull request #2719 from tejashah88/patch-1

Add ".serverless" to gitignore

bcko committed:
remove Jetbrain and MacOS related values

shiftkey committed:
Merge pull request #2718 from bcko/patch-1

Update Dart.gitignore

jayvdb committed:
Trim trailing whitespace

bamapookie committed:
Ignore usage statistics file

.idea/misc.xml used to contain usage statistics.  This was broken out to usage.statistics.xml in the latest EAP (2018.2 EAP).  It should be excluded from git.

See https://youtrack.jetbrains.com/issue/IDEA-192913

shiftkey committed:
Merge pull request #2734 from bamapookie/patch-1

Ignore usage statistics file

codethirteen committed:
Updated for Codekit 3

shiftkey committed:
Merge pull request #2728 from jayvdb/trim-trailing-spaces

Trim trailing whitespace

shiftkey committed:
Merge pull request #2694 from shastriUF/patch-1

Ignore the .cache folder for LabVIEW projects

shiftkey committed:
Merge branch 'master' into patch-1

shiftkey committed:
Merge pull request #2693 from jefrydco/patch-1

Add exception for nuxt.js project build output

shiftkey committed:
Merge pull request #2686 from 14roiron/patch-1

Matlab - Exclude coder temporary folders

shiftkey committed:
Merge pull request #2649 from lukoerfer/master

Added default annotation processing directory to Eclipse.gitignore

shiftkey committed:
Merge pull request #2631 from areyer/patch-1

allow .ssh directory

shiftkey committed:
Merge pull request #2560 from TennyZhuang/patch-2

Ignore parcel-bundler cache

shiftkey committed:
Merge branch 'master' into master

shiftkey committed:
Merge pull request #2439 from meyerlasse/master

Update JetBrains.gitignore.

shiftkey committed:
Merge pull request #1646 from josefglatz/TYPO3-improvements

[Typo3] assorted changes

shiftkey committed:
Merge pull request #2570 from justinhartman/patch-5

New template file for Backup file formats

jayvdb committed:
Backup: Add *.orig

jayvdb committed:
New file for artifacts of patch

jayvdb committed:
New file for diff files

jayvdb committed:
Autotools: De-duplicate autom4te entry

jayvdb committed:
Revert "adding exception for *.spl from elsarticle.cls"

This reverts commit d878f04545852bc876c40f48e19dabbf184d052e.

jayvdb committed:
Revert "add the *.VC.db to unreal engine ignore"

This reverts commit 548fb984f7173dea0fbd572ef0ef3784019100c0.

jayvdb committed:
GWT: Remove duplicate entry

Duplicate added amongst additions in e2caec17

jayvdb committed:
Eagle: Remove duplicate entry

Duplicate added amongst additions in de2539a72

shiftkey committed:
stub a placeholder Travis config

shiftkey committed:
Merge pull request #2740 from jayvdb/dup-autom4te

Autotools: De-duplicate autom4te entry

shiftkey committed:
Merge pull request #2741 from jayvdb/de-exact-dup

Remove exact duplicates in the same file

shiftkey committed:
Merge pull request #2736 from codethirteen/patch-1

Updated for Codekit 3

gzm55 committed:
maven-wrapper.jar should be ignored

shiftkey committed:
Merge pull request #2658 from gzm55/master

maven-wrapper.jar can be removed from the source

shiftkey committed:
Merge pull request #2153 from briandfoy/master

A new template for Perl 6

olets committed:
Craft: update to follow official recs; add 'not necessary in v3' note

shiftkey committed:
Merge pull request #2744 from olets/craft

update Craft CMS gitignore to follow Craft team's recommendations

bamapookie committed:
JetBrains: Optional excludes for auto-import

Add extra commented section to use when using Gradle or Maven auto-import.
These are mentioned in the original reference for optional excludes (https://intellij-support.jetbrains.com/hc/en-us/articles/206544839).
If this would be better as a separate, non-commented ignore file, let me know and I will resubmit.

bamapookie committed:
Remove duplicate libraries entry

fidmor89 committed:
Update Swift.gitignore

Adding a comment in case one wants to ignore .xcworkspace

fidmor89 committed:
Update Objective-C.gitignore

Adding a comment in case one wants to ignore .xcworkspace

shiftkey committed:
Merge pull request #2749 from fidmor89/patch-2

Update Objective-C.gitignore

shiftkey committed:
Merge pull request #2748 from fidmor89/patch-1

Update Swift.gitignore

fidmor89 committed:
Update Swift.gitignore

Adding Code Injection

bamapookie committed:
Separate module directory from project iml exclude

shiftkey committed:
Merge pull request #2747 from bamapookie/patch-2

JetBrains: Optional excludes for auto-import

iSWORD committed:
updated jetbrains gitignore to ignore contentModel

contentModel.xml is automatically generated and can be safely ignored.

Source: https://rider-support.jetbrains.com/hc/en-us/community/posts/115000661950-A-lot-of-files-in-contentModel-xml?page=1#community_comment_115000538984

shiftkey committed:
Merge pull request #2738 from jayvdb/patch-artifacts

Patch artifacts

AndyHee committed:
Update TeX.gitignore

Added LyX editor backup file extension: i,e .lyx~

AndyHee committed:
Update TeX.gitignore

shiftkey committed:
Merge pull request #2759 from AndyHee/master

Update TeX.gitignore

EmreAtes committed:
Add configure logs

These files are created when running `autoreconf`; however they don't need to be added to version control.

shiftkey committed:
Merge pull request #2762 from EmreAtes/patch-1

Add configure logs

ofthelit committed:
Ignore files from Local History plug-ins

Local History for Visual Studio automatically creates a history of your files every time they are saved. The history can be compared with the current version. It is a solution situated between version control and undo/redo of history, available by default in Eclipse and IntelliJ IDEs.
https://marketplace.visualstudio.com/items?itemName=AronDCurzon.LocalHistoryforVisualStudio
https://marketplace.visualstudio.com/items?itemName=lostalloy.LocalHistory-for-Visual-Studio

shiftkey committed:
Merge pull request #2768 from ofthelit/master

Ignore files from Local History plug-ins

BlackStarX42 committed:
Fixing MIDL header file pattern for Visual Studio

_i.h given in 799ee6b79e1d32f49ded6683b5364017fd4ee0a5#gitext://gotocommit/799ee6b79e1d32f49ded6683b5364017fd4ee0a5 is wrong

shiftkey committed:
Merge pull request #2769 from BlackStarX42/master

Fixing MIDL header file pattern for Visual Studio

fabiofdsantos committed:
Update Laravel.gitignore

shiftkey committed:
Merge pull request #2772 from fabiofdsantos/patch-1

Update Laravel.gitignore

StefanKrnajski committed:
Ignore logs for Symfony4

Source: https://symfony.com/doc/master/logging.html
Under Where Logs are Stored, added since Symfony 4.0

shiftkey committed:
Merge pull request #2739 from jayvdb/diff-files

New template for diff files

shiftkey committed:
Merge pull request #2758 from iSWORD/patch-1

updated jetbrains gitignore to ignore contentModel

shiftkey committed:
Merge pull request #2775 from TheGamingBook/ignore-logs-symfony4

Ignore logs for Symfony4

shiftkey committed:
Merge pull request #2751 from fidmor89/patch-1

Update Swift.gitignore

hadisfr committed:
update boilerplates to ea28c14

https://github.com/github/gitignore/commit/ea28c14da0faf75047165c10223635ba95566ad7

hadisfr committed:
fix "AttributeError: 'str' object has no attribute 'decode'" Exception

fix https://github.com/kevinxucs/Sublime-Gitignore/pull/7#issuecomment-412330884

hadisfr committed:
fix PEP8 N801 warning

hadisfr committed:
set syntax

hadisfr committed:
add a new line to the end of result

hadisfr committed:
sort list of files

hadisfr committed:
change " to '

hadisfr committed:
refactor

hadisfr committed:
add support of nested directories

hadisfr committed:
use submodule for boilerplates

hadisfr committed:
update readme to be synced with aa250e9

hadisfr commented:
The following boilerplates from https://github.com/kevinxucs/Sublime-Gitignore/commit/9b87ee16d470c68eda2f2fea4eadb1aa6a1a32f5 seems to be missing from https://github.com/github/gitignore/commit/ea28c14da0faf75047165c10223635ba95566ad7:
* [Bancha.gitignore](boilerplates/Bancha.gitignore) has been removed by https://github.com/github/gitignore/pull/1192
* [CSharp.gitignore](boilerplates/CSharp.gitignore) has been consolidate into [VisualStudio.gitignore](boilerplates/VisualStudio.gitignore) by https://github.com/github/gitignore/pull/492
* [Compass.gitignore](boilerplates/Compass.gitignore) has been merged to [Sass.gitignore](boilerplates/Sass.gitignore) by https://github.com/github/gitignore/commit/503bde5d249374a35ea9296ba707af66b03451db in https://github.com/github/gitignore/pull/945
* [Django.gitignore](boilerplates/Django.gitignore) has been merged into [Python.gitignore](boilerplates/Python.gitignore) by https://github.com/github/gitignore/pull/473
* [IPythonNotebook.gitignore](boilerplates/IPythonNotebook.gitignore) has been deleted due to existence of Jupyter Notebook of [Python.gitignore](boilerplates/Python.gitignore) by https://github.com/github/gitignore/pull/2172
* [IntelliJ.gitignore](boilerplates/IntelliJ.gitignore) appears in [JetBrains.gitignore](boilerplates/Global/JetBrains.gitignore)
* [Jython.gitignore](boilerplates/Jython.gitignore) has been merged into [Python.gitignore](boilerplates/Python.gitignore) by https://github.com/github/gitignore/pull/1493
* [LaTeX.gitignore](boilerplates/LaTeX.gitignore) has been renamed to [TeX.gitignore](boilerplates/TeX.gitignore) in https://github.com/github/gitignore/commit/81f8222e7fb5b1174b6c83016f0d6c5f28d6abea by https://github.com/github/gitignore/pull/832
* [Meteor.gitignore](boilerplates/Meteor.gitignore) has been removed by https://github.com/github/gitignore/pull/1521
* [OSX.gitignore](boilerplates/OSX.gitignore) has been renamed to [macOS.gitignore](boilerplates/Global/macOS.gitignore) by https://github.com/github/gitignore/pull/2085
* [PhPStorm.gitignore](boilerplates/PhPStorm.gitignore) appears in [JetBrains.gitignore](boilerplates/Global/JetBrains.gitignore)
* [PyCharm.gitignore](boilerplates/PyCharm.gitignore) appears in [JetBrains.gitignore](boilerplates/Global/JetBrains.gitignore)
* [Quartus2.gitignore](boilerplates/Quartus2.gitignore) has been removed by https://github.com/github/gitignore/pull/1193
* [RubyMine.gitignore](boilerplates/RubyMine.gitignore) appears in [JetBrains.gitignore](boilerplates/Global/JetBrains.gitignore)
* [SASS.gitignore](boilerplates/SASS.gitignore) has been renamed to [Sass.gitignore](boilerplates/Sass.gitignore) by https://github.com/github/gitignore/commit/503bde5d249374a35ea9296ba707af66b03451db in https://github.com/github/gitignore/pull/945
* [Symfony2.gitignore](boilerplates/Symfony2.gitignore) has been renamed to [Symfony.gitignore](boilerplates/Symfony.gitignore) in https://github.com/github/gitignore/commit/5e34ab0bc2c0aebc0af0ee1419671043734fc346 by https://github.com/github/gitignore/pull/1117
* [Target3001.gitignore](boilerplates/Target3001.gitignore) has been removed by https://github.com/github/gitignore/pull/1190
* [Tasm.gitignore](boilerplates/Tasm.gitignore) has been removed by https://github.com/github/gitignore/pull/1189
* [VB.Net.gitignore](VB.Net.gitignore) has been consolidate into [VisualStudio.gitignore](boilerplates/VisualStudio.gitignore) by https://github.com/github/gitignore/pull/492
* [vim.gitignore](boilerplates/vim.gitignore) has been renamed to [Vim.gitignore](boilerplates/Global/Vim.gitignore)
* [webMethods.gitignore](boilerplates/webMethods.gitignore) has been removed in https://github.com/github/gitignore/commit/9f1b7d77e24ec9c3e2681da93b68767fd96dd52b by https://github.com/dereknex/gitignore/pull/1

Since I have used some of deleted ones recently myself, I'll try to add ability to read _.gitignore_ files from a folder under _Packages/User_ directory.

It may be a good idea to include this post or a link to it to Package Control change log.

hadisfr committed:
remove legacy things remaining from old gibo-dependent version

Ref: https://github.com/kevinxucs/Sublime-Gitignore/commit/bc6e0dca423fb644ff27413b43e8c9d46cd05116

hadisfr committed:
Create CHANGELOG.md

hadisfr committed:
add support for user-defined boilerplates

Ref: https://github.com/kevinxucs/Sublime-Gitignore/pull/8#issuecomment-412350729

hadisfr committed:
add comments about old-gitignores

Ref: https://github.com/hadisfr/old-gitignores

hadisfr committed:
add comment about webMethods.gitignore

hadisfr committed:
fix a bug introducesd in 86b3ac0 preventing command from appearing in menu

Ref: https://github.com/hadisfr/Sublime-Gitignore/commit/86b3ac0264b6d22e34766e60acc92e69f1e495ee

kevinxucs commented on README.md:
````diff
@@ -1,42 +1,53 @@
-```
-git clone https://github.com/theadamlt/Sublime-Gitignore ~/.config/sublime-text-2/Packages/Gitignore
+```bash
+git clone https://github.com/kevinxucs/Sublime-Gitignore ~/.config/sublime-text-2/Packages/Gitignore
````
probably can change to path for sublime text 3 here

kevinxucs commented on README.md:
```diff
@@ -1,42 +1,53 @@
+[Install Package Control](https://packagecontrol.io/installation),
+then press `Cmd + Shift + P` for Mac and `Ctrl + Shift + P` for Linux/Windows,
+select `Package Control: Install Package`,
+and select `Sublime-Gitignore`.
```
`Sublime-Gitignore` -> `Gitignore`

kevinxucs commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "boilerplates"]
```
Could you please make sure when installed through package control, it downloads the submodules?

hadisfr replied:
I didn't know how to do it. Can you tell me how to do it? Is it only a regular zip file with specific extension in `Installed Packages` folder?

hadisfr replied:
It seems there are some issues here….
https://github.com/wbond/package_control/issues/126#issuecomment-37152975

hadisfr replied:
I removed submodule (https://github.com/kevinxucs/Sublime-Gitignore/pull/8/commits/497b28d1e9bc05e961dcafeb6add83c7d7e63b00) and added subtree (https://github.com/kevinxucs/Sublime-Gitignore/pull/8/commits/7980d1f691cbb55533cb348388d6ffdfe99e1b70). There will be no auto update as I wished to be, but the [boilerplates](https://github.com/kevinxucs/Sublime-Gitignore/tree/master/boilerplates) is even with the [source repo](https://github.com/github/gitignore) now, containing the full history and having the same structure. And there is some more improvements in this branch, e.g. https://github.com/kevinxucs/Sublime-Gitignore/pull/8/commits/5b35d778d2f4d21148016bca080029af7953d8ab.

I don't now there's any reason to use subtree instead of the old structure or no. I prefer the new one personally. But there is also https://github.com/kevinxucs/Sublime-Gitignore/pull/7 and it's up to you to decide.

hadisfr committed:
change installation instructions to ST3

Ref: https://github.com/kevinxucs/Sublime-Gitignore/pull/8#discussion_r209830845

hadisfr committed:
fix package name in package control

Ref: https://github.com/kevinxucs/Sublime-Gitignore/pull/8#discussion_r209831078

hadisfr committed:
decode binray output of zip package

Ref: https://github.com/hadisfr/Sublime-Gitignore/commit/24484cc3151c23400058afbcdb1b0b1a6b4d3280#r30066711

hadisfr committed:
remove submodule

hadisfr committed:
Add 'boilerplates/' from commit 'ea28c14da0faf75047165c10223635ba95566ad7'

git-subtree-dir: boilerplates
git-subtree-mainline: 497b28d1e9bc05e961dcafeb6add83c7d7e63b00
git-subtree-split: ea28c14da0faf75047165c10223635ba95566ad7

hadisfr committed:
add guide to update boilerplates

hadisfr commented:
I think it will break on windows. `os.path.join` and similar functions should be used to keep code platform-independent.

hadisfr committed:
use platform-independent path separator

fix https://github.com/kevinxucs/Sublime-Gitignore/pull/8#issuecomment-413308233

hadisfr committed:
remove submodule from README

hadisfr commented:
@kevinxucs What should we do now?

hadisfr committed:
update boilerplates to '8a6608a3f50055b99f0c9bf93ba70d0e7fa1d454'

hadisfr committed:
Squashed 'boilerplates/' changes from ea28c14..f9291de

f9291de KiCAD: the -rescue.lib file is also precious! (#3025)
a00cb87 Revert "KiCad: add footprint cache (#3043)" (#3075)
c1faae4 Add db.sqlite3-journal to Python.gitignore (#3063)
6539838 Make ignoring .ist files for glossaries-extra ... (#3066)
310f125 style: (Golang) Remove the non-essentials (#3061)
cdab24d vim: add extra session file Sessionx.vim (#3058)
6cb96f4 add the ist file generated by glossaries-extra package (#3056)
74643fe Fix log filename for Vagrant (#3053)
837181e Updated Xcode.gitignore template with a Patch (#2566)
b5aa92c Add ignores for file-based projects (#2902)
679f51c [Android] Change to Android Studio 3 (#3004)
7c462a3 Added Preamble (#2954)
445fddf Updated ignored list (#2972)
d82e9e9 NUnit - reorganize (#2960)
e58b3e0 Update Node.gitignore (#2934)
3e263d4 [Android] ignore .idea/navEditor.xml (#3012)
d44a0cb Add fp-info-cache to KiCad gitignore (#3013)
65bc8a7 [Unity] ignore Jetbrains plugin folder (#3026)
5808b77 Update Windows.gitignore (#2191)
79d46ba Add .tsbuildinfo files to Node.gitignore (#3052)
b2d72bf Exclude memory profiler captures from git (#3049)
c975d35 Replace non-ASCII character in comment (#3050)
5c5a1b5 ignore vendor directory (#3033)
8f83007 Update VisualStudio.gitignore to add a safer rule for rdl backups (#3047)
5b50e4c Updating Swift.gitignore with accio folders (#3046)
6ffcee9 KiCad: add footprint cache (#3043)
65bb94d Update VisualStudio.gitignore to ignore NuGet Symbol Packages (#3041)
293863c Delete .ist from TeX.gitignore (#3040)
26f49b0 Update R.gitignore (#3034)
c1b7904 gitignore for JENKINS_HOME Jenkins settings (#1763)
7eef17f Ignore transpiled .java and .class files (#3016)
56e3f5a [Unity] Added leading slashes to ignored directories so that valid subdirectories aren't ignored incorrectly (#2980)
6d467f5 Update OCaml.gitignore (#3010)
7f765e3 Remove wwwroot/lib exclusion for ASP.NET Core (#3009)
91cca65 Update SCons.gitignore (#3001)
cf8aca5 Update Node.gitignore (#3002)
76512d9 Add more standard ignored files for Julia (#2992)
1a417fe Don't ignore tmp and log directory (#2999)
bbd09f3 Add more Windows Store app package types (#2995)
f1f5397 docs: update Terraform.gitignore (#2998)
503c26b Add lerna-debug.log to Node.gitignore (#3000)
2aabd7b Ignore files generated by luatexja-ruby (#2996)
b960d3c Update VisualStudio.gitignore file (#2975)
96baa63 Remove Rider (idea) ignore statements from VS (#2991)
7079791 Add ignore rules for Pipenv (#2977)
23f7a81 Unity: ignore TextMesh Pro (#2988)
e71843a Move Drupal7 to correct location (#2985)
1436b29 [Node] Ignore diagnostic report files (#2971)
37abd93 Ignore Drupal test generated files (#2982)
d7452bf Add MAlonzo directory. (#2978)
dfc5642 Add .phpunit.result.cache to the .gitignore (#2973)
cd49307 Remove extra newline (#2983)
e32de69 Merge pull request #2974 from kaasasolut/patch-1
210a49a R: ignore user-specific files
d2c1bb2 Merge pull request #2963 from OsirisTerje/MigrationBackup
8dff1be Added backup folder for Package Reference Convert tool in VS 2017
66f2cae Merge pull request #1 from github/master
d34e915 Merge pull request #2957 from happydude/patch-1
d72f17f Update VisualStudio.gitignore
f731569 Merge pull request #2955 from martonmiklos/add_qtcreator_import_project_files
1ee3d75 Add *.creator.user* files to the Qt .gitignore
2fa8002 Merge pull request #2948 from Yousha/master
8372d78 Merge pull request #2952 from KexyBiscuit/patch-1
4e9a58b Remove trailing whitespace added last commit.
c2a98e4 Update CodeSniffer.gitignore
042be9d Merge pull request #2927 from Yousha/master
e7fa74b Merge pull request #2936 from martonmiklos/add_compilation_db_to_qt_gitignore
86a1e39 Merge pull request #2944 from pappasam/Python/pip-wheel-metadata
866572d Add pip-wheel-metadata to Python.gitignore
46c9c48 Merge pull request #2942 from joaomlneto/patch-1
f273474 Ignore CMake FetchContent directory
7b39587 Merge pull request #2939 from Flate/master
28ae73e Unity: Never ignore asset meta data
3818fec Merge pull request #2938 from locofocos/patch-1
dfce3fd Update Ruby.ignore to ignore .byebug_history file
8338f33 Merge branch 'master' of https://github.com/github/gitignore into add_compilation_db_to_qt_gitignore
1c1744a Added compile_commands.json to Qt.gitignore
f908e51 Merge pull request #2925 from kaiyangjia/master
ec37913 Add build_file_checksums.ser and modules.xml files.
20f0ea9 Merge pull request #2926 from github/mdbs-are-people-too
152f419 Unity: Do not blanket-ignore specific plugins, that's a user choice
24f6734 Unity: ignore mdb files and fix path to crashlytics file (it can be in other places)
240dd3f add android lint files
f549603 Merge pull request #2922 from wRorsjakz/patch-1
df47060 Merge pull request #2921 from aslamanver/master
2b2b613 Add version control ignore
584dd34 google-services.json
18654b4 Merge pull request #2918 from johnsoncodehk/patch-3
81923b1 Unified base folder name format
6531e38 Merge pull request #2899 from johnsoncodehk/patch-2
d1baec1 Merge pull request #2907 from HassanHashemi/patch-1
98a3e85 cleanup
d7eef7d Merge pull request #2913 from dangu/dev/IAR_EWARM
5bbde8f Update the rule
db53e8c Changed the rule to healthchecksdb
1afdbc0 Move to community/embedded
c7d0718 Update info for IAR EWARM
718f4f5 Move IAR_EWARM to a new directory Embedded
c5557ee Move to the community directory
ade77e5 Adding .gitignore for IAR EWARM IDE (https://www.iar.com/iar-embedded-workbench/)
6cc2079 Add * wildcard
21780d9 Merge pull request #2912 from FutureClemens/Unity_Logs
f3d0de0 Added Logs to Unity.gitignore, which is automatically generated in Unity 2018.3 to log package updates.
e35762d Merge pull request #1 from github/master
a53b4a3 Merge pull request #2896 from erdtsksn/master
745923d Merge pull request #2889 from rubberduck203/atmelstudio
ac2e28a Merge pull request #2908 from lmedson/env_test_ignore
4e50d1c Merge pull request #2894 from colearendt/remove-rsconnect
e5f7d39 Merge pull request #2893 from rafaelstz/patch-8
65c5637 Merge pull request #2895 from tchajed/go-fix-typo
e67bbf5 Merge pull request #2900 from theDrake/patch-1
6cbcdee Merge pull request #2904 from wldevries/VisualStudio_CacheDirs
1738bf6 Merge pull request #2905 from marcusolini/netbeans-gen-files
ddf99c1 Merge pull request #2906 from ksmit799/patch-1
3be56ed Merge pull request #2890 from stevenleeS0ht/master
5abe139 Remove empty line
2861abc Add BeatPulse specific section
9846160 Merge pull request #2910 from kesoji/master
c2d0605 fix wsp
f966096 Add ignorer to .env.test variable
b2310e0 Ignore health check temp database
32fa57b [Unity] Ignore .gradle/ build cache
2e5099d macOS low cap m
2731b70 Ignore Generated Files
38b9fdf Only unignore directories that have at least a character ending with .cache
47d29f3 Improve comment capitalization consistency
8e6fda2 Ignore Logs/Packages-Update.log
fd3fe58 feat: Add missing executables and generated
15736d4 [Go] Fix typo in comment
512393f Create Magento2.gitignore
9771d5c remove rsconnect folder from R.gitignore
941ea22 correct capitalization problems and add Octave which link to MATLAB
8b335ee Create AtmelStudio.gitignore
e313e29 Merge pull request #2307 from marcrocny/marcrocny-patch-1
83a01cf Merge pull request #2885 from github/xilinux-template
541a7ae added template for Xilinx
8c943e0 Merge pull request #2376 from vuolter/patch-2
5ec4476 Merge pull request #2027 from rejurime/dolphin-smalltalk
c017c47 Merge pull request #1731 from christophmuellerorg/Virtuoso
79d9eb9 Merge pull request #2861 from Coeur/revert-2752-patch-2
6615b26 Merge pull request #2874 from 1nfinite9/patch-1
eb14c04 Merge pull request #2884 from github/more-community-cleanup
056eea8 added template for Splunk apps
40522d5 added template for Puppet modules
45aaaf2 added template for Nikola framework
db803b4 add template for Jupyter Notebooks
2ee0576 added template for Exercism project
d6d2a41 added template for Bazel
6fd7cae Merge pull request #2849 from venatoria/master
e41d887 Merge pull request #2857 from github/experiment-ecosystem-groupings
f7c4f2e line endings
e9883e5 update header for Drupal template
b7dc63a added bitrix community template
9c47778 Update README.md
f90299c Merge pull request #2877 from skident/master
1548cc4 Merge pull request #2871 from cwize1/AddArmToVisualStudio
4a8e0a1 Merge pull request #2875 from watilde/emacs-network
9b35471 Merge pull request #2880 from TwoBitAlchemist/master
926a6cb Hide wheels directory used on Debian-like systems (Ubuntu)
3352aa4 Merge branch 'master' into experiment-ecosystem-groupings
c2bb55b Merge pull request #2879 from cailloumajor/patch-1
da37cec [TeX] Ignore generated files by 'vhistory' package
36ea81b Added CMakeLists.txt.user file to a CMake .gitignore
87ec1bd Add "network-security.data"
58f322c Update Unity.gitignore
f2bd62d Merge pull request #2872 from rhysforyou/elixir-ls
1cdba03 ignore files generated by ElixirLS
b3d1e72 Add `ARM` and `ARM64` to Visual Studio's ignore file.
0a23ed5 Revert "Update Xcode.gitignore"
2277645 add community template for NW.js
d9977d1 add community template for Cordova
f29f404 add community template for LogTalk
e5c5a2b add community template for Racket
213a053 added community template for PHP CodeSniffer
7637178 add community template for Meteor
eff87b9 added community template for InforCRM
991acca add community template for KenticoCMS
8a2fae7 add templates for older JBoss versions
e67a985 Add community template for Red language
d298644 added Hugo community template
bbd34a4 add community Snap template
2055dc1 Added community template for ThinkPHP
bd97b79 added community Pimcore template
05c5eac add a template for Phoenix projects
1ab24a3 add support for Nuxt.js
eda6ba5 rename ecosystem to community
c440ae4 stub an example Vue gitignore template
3641b16 tweaks to example template
ded1dab a bit more edits
1abbddb expanded new sections
1980e4f prettify things
37ec8ba add section about updated folder structure
663ff83 Move Magento v1 into ecosystem
7af655e add a better header to the old template
344e52b Update default template to Drupal 8
db1d52b Move current Drupal into ecosystem directory
ec24607 Merge pull request #2819 from AurelienLavorel/AurelienLavorel-patch-1
68e04fd Merge pull request #2569 from justinhartman/patch-4
f1e6f0d Merge pull request #2086 from AlexanderBrevig/master
ee2ae04 No need for .vscode line
2384243 Merge pull request #2847 from FutureClemens/Unity_VC.db_Update
32c358b Merge pull request #2851 from kitsuyui/add-pyre-directory
e970e56 Add .pyre directory for Pyre type checker
b75ca86 vscode, dynamodb, serverless folders
45dca6a Merge pull request #2848 from paxperscientiam/patch-1
5e89a98 Add support for FuseBox
a2e5aea Added *.VC.db to Unity.gitignore, which is an auto generated IntelliSense file from VS2015 Update 2 (replacing the .sdf file).
c68eb91 Merge pull request #2752 from fidmor89/patch-2
5f548f8 Merge pull request #2805 from lakerobin/patch-1
13c6410 Merge pull request #2806 from Orbyt/patch-1
93d97e7 Merge pull request #2837 from chhh/feature/netbeans-patch
6af9831 Ignoring multi-directory a path without ** is not enough for nested projects
2a40ae6 Merge pull request #2825 from snakeice/patch-1
64df318 Merge pull request #2835 from dureyingenieria/ignore-android-bundles
46f2003 Update Delphi.gitignore
053f167 Merge pull request #2829 from valentine/rails-public-assets
5e2428f Merge pull request #2830 from kevdotbadger/patch-1
95ba18c Ignore the new Android Bundle generated files
372d6fd Update README.md
86fdc70 Ignore public/assets in Rails .gitignore
4fff0ee Update Delphi.gitignore
5002bec New dependency manager delphi for delphi
72190ee Merge pull request #2821 from KevinRaney/add-to-visual-studio
3f0a607 Update VisualStudio.gitignore
d6af959 Remove patch files from repo
340e2fe Merge pull request #2818 from whisperity/ignore-tex-templategenerated
a77aaae [TeX] Ignore generated files by 'comment' and 'xcolor' package
7ffe053 Add .log and .zip to ignores, Thanks Alex Bustos
cfd32d9 Merge pull request #2815 from Droni/patch-1
b0b1b52 No need to fully exclude CodeRush settings
9d7ff09 Merge pull request #2810 from stuckj/patch-1
519396a Merge pull request #2811 from JeffCyr/wpftmp
a89e5e1 Added *_wpftmp.csproj ignore
8e9a976 Ignore build cache file for Android Studio 3.1+
8025c33 Added wildcard ignore for .keystore files
28222da Merge pull request #2804 from louistio/grunt-js-https
67aa0dc Only ignore root vendor directory
ab098d2 Use https for gruntjs link in Node gitignore
35441e9 Merge pull request #2802 from SethMichaelLarson/patch-1
34e8d3e Add .nox directory for the nox testing tool
90cb995 Merge pull request #2797 from gurvircheema/rails5.2
7598e1b Merge pull request #2799 from alastairtree/patch-1
a06382d Minor imporvements to terraform ignore file
3db98a3 Add storage and yarn related entries for rails 5.2
210c95b Merge pull request #2788 from amigcamel/ignore_ipython_profile
e675c00 fix typo: should be profile_default instead of default_profile
ffff0b8 Merge pull request #2784 from GabrielC101/add_dmypy_json_to_Python.gitignore
5aff71b Merge pull request #2785 from obounaim/update-terraform
cb36216 Merge pull request #2790 from Niels-NTG/master
a53c915 Merge pull request #2781 from cnyyk/master
a92d608 Add ignore rules for Linux ARM exports
318f8fc Merge pull request #2787 from muzimuzhi/patch-1
c04384d ignore ipython profile
e98611a Added *.listing extension to TeX.gitignore
70be586 Update Terraform.gitignore
7bc8780 Remove newline at end of Python.gitignore.
eb4ff42 add dmypy.json - without initial '.', to Python.gitignore.
07a1dd6 add .dmypy.json to Python.gitignore.
38b189d Merge pull request #2782 from Peetz0r/patch-1
b7477e9 KiCad .gitignore library fix
cbdbe97 Removed Microsoft Word 2016 Auto Backuped File on macOS
f19d2a5 Update Xcode.gitignore
430b19e New template file for Images
1da4978 Create PuTTY.gitignore
6686e52 update wwwroot/lib description; revert EOF
d5724e4 VS15 bower - default config
5a355ed Add gitignore for Cypress Semiconductor IDE
236e6d0 Add support for Dolphin Smalltalk
42b5cf3 Initial gitignore file for [Cadence Virtuoso](http://www.cadence.com/products/cic/Pages/default.aspx) Cadence Virtuoso is a commercial EDA tool used for custom IC (ASIC) design. This gitignore file blacklists database lock files, the contents of run directories for layout versus schematic and design rule checks as well as log files for a selection of tools inside the package.

git-subtree-dir: boilerplates
git-subtree-split: f9291de89f5f7dc0d3d87f9eb111b839f81d5dbc

hadisfr committed:
merge commit 'bc807e6f2c5fde1b392d39aad294957374fbf118' into dev

hadisfr committed:
Squashed 'boilerplates/' changes from f9291de..218a941

218a941 KiCad: add *.kicad_sch-bak and *.kicad_prl (#3427)
f44d830 Remove pointless blank line from the beginning of the Godot file (#3466)
36ce3a8 Exclude *.tfvars files (#3452)
eea2893 ignore HPROF files (#3456)
5cf3934 ignore *~ files in the po directory (#3453)
c9fab68 Update stale heading anchor (#3445)
6eff882 Fix false positives on Coverage*.cs files (#3454)
620d487 Fixing target and adding debug to Rust gitignore (#3436)
18aa6d8 Add *.aab (#3414)
6935207 ignore .xwm generated by xwatermark package in TeX (#3412)
7293c14 Add .yarn/install-state.gz to Node.gitignore (#3407)
07b3cd7 Dart: Ignore .flutter-plugins (#3398)
eee21bf Ignore Next.js `next export` result directory (#3387)
2a4de26 Add FodyWeavers.xsd to VisualStudio.gitignore (#3380)
8058738 Ignore Gradle subproject build dir (#3370)
14f8a8b Remove pip-wheel-metadata/ from Python.gitignore (#3364)
994f99f Add generated timing files to Coq.gitignore (#3367)
2f75277 Ignore AWS Serverless Application Model build folder (#3357)
8300d88 Ignore Terraform CLI configuration files (#3359)
e607bdd Revert "[Python] adding IDE & static related stuff (#3317)" (#3363)
eefa65c Ignore Local History folder (#3008)
3df9bc0 Update R.gitignore (#3354)
e931ef7 [Python] adding IDE & static related stuff (#3317)
ae5431d Changes to PyBuilder 0.12+ (#3338)
ef59ea5 Extend Qt.gitignore with qmlcache qrc files (#3322)
c851a10 update README (#3248)
811be05 Add Win32/ to gitginore list (#3308)
c022baa Update Autotools.gitignore. (#3337)
0af689b [VisualStudio] Add rule for ScaffoldingReadMe.txt (#3328)
be4f76b [Node] Add yarn v2 excludes (#3327)
dc8c27c Add Unity's Addressables and Android's temp files. (#3311)
81ebaec Improved JENKINS_HOME example (#3332)
589831e Ignore Cython debug symbols (#3330)
823e38f Add /public/uploads for file upload gem Shrine (#3333)
4d4af94 Update Drupal.gitignore (#3329)
863f9d2 update parcel cache (#3331)
9904c12 Jigsaw (#3334)
1095d93 Update for Prestashop 1.7 (#3261)
8918d73 Add period. (#3336)
599646e Include Rider as a supported IDE (#3323)
d650516 Changed TeX to not ignore tikz files by default. (#3320)
219b125 Remote typings ignore from node gitignore (#3319)
65047ae Add ignore to umbraco folder (#3307)
111cf18 Changes the name of Perl 6 to Raku (#3312)
83ff6af Add `cover/` to Python.gitignore. (#3310)
8d88fa5 Add snowpack dependency directory (#3315)
55df35e add in global metals (#3296)
98f28b4 add alternative sftp settings file (#3006)
6e5b49b Add Umbraco V8 packages folder location (#3302)
583f11e [Python] .python-version should NOT be ignored (#3274)
7cc00a1 Update concrete5 gitignore (#3176)
7b5783a Update Unity.gitignore (#3297)
9627ac2 Ignore .pytype/ directory (#3301)
584ee18 [Global/Eclipse] Make .project opt-in (#3305)
7ab549f Update PureScript adding .spago (#3278)
0e8e8b1 [Godot] Ignore imported translations (#3269)
eed88c1 TeX: Ignore REVTeX generated Notes.bib files (#3286)
b24e6f7 Update Autotools.gitignore (#3271)
f726843 Added a comment why MemoryCaptures get excluded (#3276)
6c87d24 Ignore files inside `.vscode-test` (#3256)
b4551f2 added Coverlet result files as a Code Coverage Tool (#3252)
1e66757 [Unity] Fix ignore Jetbrains plugin folder (#3266)
8a4b278 Added gitignore for GNOME Extension (#3268)
6e7ca98 Update Node.gitignore (#3264)
21e9f06 Added jarRepositories.xml as an optional ignore (#3254)
1604c21 Add .env.* to optionally ignored files in Rails.gitignore. (#3065)
e2e54ce Modified comment on Asset meta data ignore to better explain usage. (#3249)
38c1cfc Update Coq.gitignore after coq/coq#11075 (#3247)
db0ef78 wtf is a pyflow (#3242)
1c11457 Added aar file type to gitignore (#3233)
c3e97a0 exclude compiler.xml and artifacts when using Gradle (#3236)
cb0c6ef ignore new directory added in Android Studio 3.5 (#3232)
0207a9c Ignore the __pypackages__ folder for pyflow users. (#3205)
b58b534 exclude *.tps files from TeXnicCenter (#3213)
e5b26f5 exclude .Renviron (#3214)
44bc08a Update Coq.gitignore for coq/coq#8642 and coq/coq#10947 (#3223)
b2198b3 Added celerybeat.pid (#3229)
766cf0c Add crash.log to Packer.gitignore (#3230)
cedf2d1 JustCode has been discontinued as the features it provided are now included in Visual Studio 2017 and newer. (#3227)
3794768 Update Archives.gitignore (#3220)
2644536 Comment out / add note about conflicting gitignore lines (#3204)
a14296f IPython support for jupyter-notebook gitignore files (#3211)
21419e3 Update to Global/VisualStudioCode (#3201)
475f63b Add versioned so files extension to Qt.gitignore (#3186)
b912aea Add qmake generated prl files to Qt.gitignore (#3187)
baf1355 Ruby: ignore RuboCop remote inherited config files (#3197)
e95b1be Add apt_generated_test/ (#3193)
fb4019a Added tern-port (#3195)
086eaca feat: Add microbundle cache (#3116)
3b445eb Ignore coverage annotations reports. (#3131)
e9bffd2 Added new checksums and allows for relative directories (#3128)
26c3d84 [VisualStudio] add Logs folder (#3143)
a55ad76 Add generated translations to the .gitignore (#3159)
72d88e7 Remove `release/` from Android.gitignore (#3162)
0d7600f Add rebar3 tool files and directories (#3178)
0a7dc39 Ignore GatsbyJS files (#3180)
85bf08b Updated Nim.gitignore (#3172)
3dc70c8 Upgrate to rails 6.0.0 (#3167)
aad5143 Add OpenSSL-related file extensions (#3168)
41ec058 Update Vim.gitignore (#3169)
499ae89 Added Simulink cache files (#3148)
c70562c Add esp-idf.gitignore (#3150)
0455041 Add *.sch-bak to KiCad.gitignore (#3153)
6651bca Reference why to exclude binary jar, fixes #111 (#3154)
e92f8db Update Autotools.gitignore (#3005)
e00d43a Improving the documentation of Objective-C and Swift files (#3133)
a200716 Ignore gummi previews (#3144)
a3e8e14 Swift: ignore .swiftpm (#3083)
20e9638 Update Godot.gitignore (#3142)
c4ed859 Revert of #2988. TextMesh Pro files are not auto managed (#3134)
9df8aa6 Partial revert of 837181e2f60198de3701a1f87e66aaf8e69d0e07 (#3132)
1bf740f Added more glossaries auxiliary files and makeindex logs to TeX template (#3104)
6ab0c59 .travis.yml: Trim trailing whitespace and indent (#3118)
db943c4 Add *.hie to Haskell.gitignore (#3120)
f900e7e Ignore the .ionide working folder (#3108)
50e42aa Update Laravel.gitignore (#3098)
7aec8fd Ignore *.xyd outline files generated by xy LaTeX package (#3088)
e447ea2 Updating Joomla.gitignore to Joomla! 3.9.8 (#3090)

git-subtree-dir: boilerplates
git-subtree-split: 218a941be92679ce67d0484547e3e142b2f5f6f0

hadisfr committed:
add update instructions
